### PR TITLE
Remove erroneous curlies in Xml resource 

### PR DESCRIPTION
**Pull request by mrj001**
This is addressing one of the 13 strings in issue 30218 that have faulty formatters (Sch_MinLengthGtBaseMinLength).  I'd like some feedback on this approach before I tackle the other 12.

### CODE REVIEW

File: src/libraries/System.Private.Xml/src/Resources/Strings.resx
Comment (danmosemsft) [inline src/libraries/System.Private.Xml/src/Resources/Strings.resx]: At this point it is anyone's guess what the original developer intended by these odd curlies. One possibility is that s/he planned to include the values instead of `{value}` ie 
`<value>It is an error if the derived 'minLength' facet value {0} is less than the parent 'minLength' facet value {1}.</value>`
and modifying the code to pass those in.

I think we would do this if this was an often seen string. However it has taken years for anyone to notice and report this bug, so it's apparently not a common string. I think updating them as you have done, without updating code to pass anything in, is fine.

File: src/libraries/System.Private.Xml/tests/XmlSchema/XmlSchemaSet/TC_SchemaSet_Compile.cs
Comment (danmosemsft) [inline src/libraries/System.Private.Xml/tests/XmlSchema/XmlSchemaSet/TC_SchemaSet_Compile.cs]: We do not currently localize strings, but if we did so in the future (which would certainly be optional if we did) we would need tests to pass on the localized builds, and this would not.

You can either just assert for the string to contain "minLength" -- so it's at least not some random unrelated string. Or I think it would be OK to not add a test at all, especially if you believe the functional code coverage already exists.

**Review by danmoseley — CHANGES_REQUESTED** [review]
Looks good but I would either change the test to be locale-independent, or (if there's already functional coverage) consider not bothering with a test at all. This would be acceptable for any cases where you are literally just changing the text of the string and there are no replacement parameters in it.

**Comment by danmosemsft** [comment]
Looks like test failures are unrelated - if you push your edit, it will run again. 

BTW, we prefer to set the PR title to something that mentions to issue, not a number.

File: src/libraries/System.Private.Xml/tests/XmlSchema/XmlSchemaSet/TC_SchemaSet_Compile.cs
Comment (mrj001) [reply src/libraries/System.Private.Xml/tests/XmlSchema/XmlSchemaSet/TC_SchemaSet_Compile.cs]: I am having difficulty checking the code coverage.  I can't even get most of the tests to run.  I'm pretty sure I'm doing something wrong.  Maybe you can advise me.

Here is how I'm building the runtime.  (I'm using git bash as the command line.)
`$ git clean -xdf`
`$ git fetch upstream master`
`$ git merge upstream/master`
`$ git push origin master`
`$ ./build.cmd -subsetCategory coreclr-runtime -c Checked`
`$ ./build.cmd -subsetCategory coreclr-libraries -runtimeConfiguration Checked`
`$ ./build.cmd -vs System.Private.Xml`

In Visual Studio, I press Ctrl + Shift + B to build the solution.  Only 15 of 29 projects build successfully.

The first error in the output window is:

> NuGet package restore failed. Please see Error List window for detailed warnings and errors.

However, there is no additional information in the Output window.  Additionally, building at the command line does not experience this problem.

The test projects which do not build all show this error:

> error NU1201: Project XmlCoreTest is not compatible with netcoreapp5.0 (.NETCoreApp,Version=v5.0). Project XmlCoreTest supports: net50 (.NETFramework,Version=v5.0)

I'm using Community Edition, so I'll have to check code coverage via command line testing anyway, so I tried this:
`cd src/libraries/System.Private.Xml`
`../../../.dotnet/dotnet test -p:CollectCoverage=true -p:CoverletOutputFormat=cobertura`

This results in numerous errors like this:

> Starting test execution, please wait...
>
>A total of 1 test files matched the specified pattern.
>Testhost process exited with error: It was not possible to find any compatible framework >version
>The framework 'Microsoft.NETCore.App', version '5.0.0' was not found.
 > - The following frameworks were found:
 >    2.1.11 at [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
 >     2.1.12 at [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
 >     2.1.13 at [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
 >     2.1.16 at [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
>      3.0.0-preview9-19423-09 at [C:\Program  Files\dotnet\shared\Microsoft.NETCore.App]
>      3.0.0 at [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
>      3.1.2 at [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
>      5.0.0-preview.3.20167.12 at >[C:\Users\mrj00\source\repos\runtime\.dotnet\shared\Microsoft.NETCore.App]
>You can resolve the problem by installing the specified framework and/or SDK.
>The specified framework can be found at:
>  - https://aka.ms/dotnet-core-applaunch?>framework=Microsoft.NETCore.App&framework_version=5.0.0&arch=x64&rid=win10-x64
>. Please check the diagnostic logs for more information.
>
>Test Run Aborted.

This appears to be the same error.  Any ideas what I'm doing wrong?  No coverlet.cobertura.xml file is then found anywhere in the project folders.

Comment (mrj001) [reply src/libraries/System.Private.Xml/tests/XmlSchema/XmlSchemaSet/TC_SchemaSet_Compile.cs]: Forgot to mention: I have DOTNET_ROOT set to the .NET core in the runtime/.dotnet folder.

However, when I run this from the System.Private.Xml folder
`../../../.dotnet/dotnet --version`

It still prints out 3.1.200.  I don't understand why as this is the local copy of .NET core downloaded by the build script.

Comment (mrj001) [reply src/libraries/System.Private.Xml/tests/XmlSchema/XmlSchemaSet/TC_SchemaSet_Compile.cs]: I've made some progress.
I've set DOTNET_MULTILEVEL_LOOKUP to 0, and started invoking it through the dotnet.cmd  script.

Running the tests now produces many errors like the following.  

> Starting test execution, please wait...
>
>A total of 1 test files matched the specified pattern.
>Testhost process exited with error: It was not possible to find any compatible framework version
>The framework 'Microsoft.NETCore.App', version '5.0.0' was not found.
>  - The following frameworks were found:
>      5.0.0-preview.3.20167.12 at [C:\Users\mrj00\source\repos\runtime\.dotnet\shared\Microsoft.NETCore.App]
>You can resolve the problem by installing the specified framework and/or SDK.
>The specified framework can be found at:
>  - https://aka.ms/dotnet-core-applaunch?framework=Microsoft.NETCore.App&framework_version=5.0.0&arch=x64&rid=win10-x64
. Please check the diagnostic logs for more information.
>
>Test Run Aborted.

Where are the "diagnostic logs"?  Why is this framework not acceptable for version '5.0.0'?  It is the internally bundled framework, so I expect it should be acceptable.

**Comment by mrj001** [comment]
@danmosemsft 
Sorry it took so long to get back to this.  I was blocked on trying to generate the code coverage report.  Finally gave up and filed #34152 .  How does one determine the code coverage with coverlet?

Since I don't have this information, I've decided to proceed with unit tests.  I suspect that if unit tests did cover these strings, they would have had proper formatters in them.  Also, it confirms to me that I understand exactly when these are being used.

In looking at Sch_LengthGtBaseLength, I have come up with two questions.  This resource is only used in method CompileLengthFacet in file FacetChecker.cs:

```
                if ((_baseFixedFlags & RestrictionFlags.Length) != 0)
                {    
                    if (!_datatype.IsEqual(_datatype.Restriction.Length, _derivedRestriction.Length))
                    {    
                        throw new XmlSchemaException(SR.Sch_FacetBaseFixed, facet);
                    }    
                }    
                if ((_baseFlags & RestrictionFlags.Length) != 0)
                {    
                    if (_datatype.Restriction.Length < _derivedRestriction.Length)
                    {    
                        throw new XmlSchemaException(SR.Sch_LengthGtBaseLength, facet);
                    }    
                }    

```
I can trigger the exception to throw with the message SR.Sch_LengthGtBaseLength by specifying a length facet in both a base type and a derived type, with the derived type having a longer length.

Q1: Isn't the length facet supposed to specify the exact length?  If so, why is the second test above checking for "less than" rather than "not equal"?
Q2: what is _baseFixedFlags vs. _baseFlags?  The first test was not triggered by my test case, but the second one was.  I'm curious as to why I'm not triggering the first test.

This is the test schema that I use to provoke the exception. 
```
<?xml version='1.0' encoding='utf-8' ?>
<xs:schema elementFormDefault='qualified'
           xmlns:xs='http://www.w3.org/2001/XMLSchema'>
    <xs:simpleType name='foo'>
        <xs:restriction base='xs:string'>
            <xs:length value='5'/>
        </xs:restriction>
    </xs:simpleType>
    <xs:simpleType name='bar'>
        <xs:restriction base='foo'>
            <xs:length value='6'/>
        </xs:restriction>
    </xs:simpleType>
</xs:schema>
```

**Comment by danmosemsft** [comment]
@mrj001 are the instructions in https://github.com/dotnet/runtime/blob/master/docs/workflow/building/libraries/code-coverage.md not working? If not @ViktorHofer owns them.

For your other questions, @buyaa-n owns System.Xml and can help.

File: src/libraries/System.Private.Xml/tests/XmlSchema/XmlSchemaSet/TC_SchemaSet_Compile.cs
Comment (danmosemsft) [inline src/libraries/System.Private.Xml/tests/XmlSchema/XmlSchemaSet/TC_SchemaSet_Compile.cs]: note you can replace all of this code with something like
```c#
Exception ex = Assert.Throws<XmlSchemaException>(()=> ss.Compile());
Assert.Contains("minLength", ex.Message);
```
or just oneline
```c#
Assert.ThrowsContains<XmlSchemaException>(()=> ss.Compile(), "minLength");
```

**Comment by buyaa-n** [comment]
> Q1: Isn't the length facet supposed to specify the exact length? 

Yes, but only for that type i assume (in this case the base type)

> If so, why is the second test above checking for "less than" rather than "not equal"?

Because derived type can have length less than or equal to base type length, why do you think it must be equal? 

> Q2: what is _baseFixedFlags vs. _baseFlags? 

Looks _baseFixedFlags is for only inferring if base type is fixed (which can be a boolean), but _baseFlags has RestrictionFlags of base (Length, MinLength, MaxLength ... etc)

> The first test was not triggered by my test case, but the second one was. I'm curious as to why I'm not triggering the first test.

I assume you need to set fixed value for base type definition for example:
`<xs:simpleType name='foo' type="xs:string" fixed="FixedValue">`

File: src/libraries/System.Private.Xml/src/Resources/Strings.resx
Comment (buyaa-n) [reply src/libraries/System.Private.Xml/src/Resources/Strings.resx]: Agree with Dan, plus looks better for me without `facet`, e.g `It is an error if the derived 'minLength' value {0} is less than the parent 'minLength' value {1}`. From the code section you mentioned of the calling routine you cans pass those values directly, something like `throw new XmlSchemaException(SR.Sch_MinLengthGtBaseMinLength, _derivedRestriction.MinLength, _datatype.Restriction.MinLength);`

**Comment by mrj001** [comment]
@buyaa-n 
> Because derived type can have length less than or equal to base type length, why do you think it must be equal?
I initially relied on this page's description:
[https://www.w3schools.com/xml/schema_facets.asp](url)
A derived type would fail the "Is A" test if its length were different from that of its base class.

For greater surety, see the note in paragraph 4.3.1.1 of:
[https://www.w3.org/TR/2012/REC-xmlschema11-2-20120405/datatypes.html](url)

In Paragraph 4.3.1.4 of the same link:

> Schema Component Constraint: length valid restriction
It is an ·error· if length is among the members of {facets} of {base type definition} and {value} is not equal to the {value} of the parent length.

And a unit test (failing) to demonstrate.  Shall I file a separate issue?  This is getting off topic of this pull request.
```
        [Fact]
        public void LengthLtBaseLength_Throws()
        {
            string schema = @"<?xml version='1.0' encoding='utf-8' ?>
<xs:schema elementFormDefault='qualified'
           xmlns:xs='http://www.w3.org/2001/XMLSchema'>
    <xs:simpleType name='foo'>
        <xs:restriction base='xs:string'>
            <xs:length value='7'/>
        </xs:restriction>
    </xs:simpleType>
    <xs:simpleType name='bar'>
        <xs:restriction base='foo'>
            <xs:length value='6'/>
        </xs:restriction>
    </xs:simpleType>
</xs:schema>
";

            XmlSchemaSet ss = new XmlSchemaSet();
            ss.Add(null, XmlReader.Create(new StringReader(schema)));

            Exception ex = Assert.Throws<XmlSchemaException>(() => ss.Compile());
            Assert.Contains("length", ex.Message);
        }

```

**Comment by buyaa-n** [comment]
> Schema Component Constraint: length valid restriction
It is an ·error· if length is among the members of {facets} of {base type definition} and {value} is not equal to the {value} of the parent length.

I think that is valid only for `fixed length`:
image

**Comment by buyaa-n** [comment]
> And a unit test (failing) to demonstrate. Shall I file a separate issue? This is getting off topic of this pull request.

By counting my above comment your test should fail unless it has `fixed = true` attibute

**Comment by mrj001** [comment]
@buyaa-n 
This is the note from Paragraph 4.3.1.1 (emphasis added):

> Note: The {fixed} property is defined for parallelism with other facets and for compatibility with version 1.0 of this specification. But it is a consequence of length valid restriction (§4.3.1.4) that the value of the length facet cannot be changed, **regardless of whether {fixed} is true or false.**

**Comment by buyaa-n** [comment]
> Note: The {fixed} property is defined for parallelism with other facets and for compatibility with **version 1.0** of this specification. But it is a consequence of length valid restriction (§4.3.1.4) that the value of the length facet cannot be changed, regardless of whether {fixed} is true or false.

@mrj001 that is good point, I understand and really appreciate your effort for trying to make this right, but FYI we only support version 1.0

**Comment by CoffeeFlux** [comment]
Think a merge got messed up somewhere @mrj001

**Comment by mrj001** [comment]
@CoffeeFlux I wonder if I did a merge instead of a rebase?  Is there something I should do about it?

**Comment by danmosemsft** [comment]
I think we just discovered one downside of CODEOWNERS files - we have a lot of reviewers for this PR now 😈 

@mrj001 I am not sure what happened, but this may help (I'm assuming that your remote for dotnet/runtime is named `upstream`, ie., `git remote -v` shows `upstream        https://github.com/dotnet/runtime (fetch)`:
```
git checkout Issue30218
git fetch upstream
git rebase upstream/master
```
then if you are happy with the result, `git push -f`

I pulled your branch down and performed these steps and seemed to have a good result.

**Comment by mrj001** [comment]
@danmosemsft I have done those steps, and the result looked good.  I checked the various resources against my list of which ones I've done & haven't done.  It all checked out.  So I've force pushed again.

I looked through my history, but did not see any evidence I accidentally did a merge instead of a rebase.  So I don't know what happened either.

File: src/libraries/System.Private.Xml/src/Resources/Strings.resx
Comment (buyaa-n) [inline src/libraries/System.Private.Xml/src/Resources/Strings.resx:1723]: Could you word this and below message similarly,  for me this version sounds better

File: src/libraries/System.Private.Xml/tests/XmlSchema/XmlSchemaSet/TC_SchemaSet_Compile.cs
Comment (buyaa-n) [inline src/libraries/System.Private.Xml/tests/XmlSchema/XmlSchemaSet/TC_SchemaSet_Compile.cs:159]:
```suggestion
        public void MinLengthLtBaseMinLength_Throws()
```

File: src/libraries/System.Private.Xml/tests/XmlSchema/XmlSchemaSet/TC_SchemaSet_Compile.cs
Comment (buyaa-n) [inline src/libraries/System.Private.Xml/tests/XmlSchema/XmlSchemaSet/TC_SchemaSet_Compile.cs:184]: NIT: extra line

File: src/libraries/System.Private.Xml/tests/XmlSchema/XmlSchemaSet/TC_SchemaSet_Compile.cs
Comment (buyaa-n) [inline src/libraries/System.Private.Xml/tests/XmlSchema/XmlSchemaSet/TC_SchemaSet_Compile.cs:381]: Please change casing of literal `minlength` instead of converting message to lower case

File: src/libraries/System.Private.Xml/tests/XmlSchema/XmlSchemaSet/TC_SchemaSet_Compile.cs
Comment (buyaa-n) [inline src/libraries/System.Private.Xml/tests/XmlSchema/XmlSchemaSet/TC_SchemaSet_Compile.cs:385]: Is this for checking curly braces? I don't really think we need to test that, just removing them from resource file would be enough

File: src/libraries/System.Private.Xml/tests/XmlSchema/XmlSchemaSet/TC_SchemaSet_Compile.cs
Comment (buyaa-n) [inline src/libraries/System.Private.Xml/tests/XmlSchema/XmlSchemaSet/TC_SchemaSet_Compile.cs:389]: This seems success edge cases, thanks for adding this, as we have `MaxMinLengthBaseLength_ThrowsData` above could rename it that express its for success cases

File: src/libraries/System.Private.Xml/tests/XmlSchema/XmlSchemaSet/TC_SchemaSet_Compile.cs
Comment (buyaa-n) [inline src/libraries/System.Private.Xml/tests/XmlSchema/XmlSchemaSet/TC_SchemaSet_Compile.cs:587]: Now i see it is not testing exception thrown, its testing successful compilation, in that case you don't need to use try catch, just compile the schema set, if there is error the test will fail anyways. Maybe add Assert.True(true); at the end

File: src/libraries/System.Private.Xml/tests/XmlSchema/XmlSchemaSet/TC_SchemaSet_Compile.cs
Comment (buyaa-n) [inline src/libraries/System.Private.Xml/tests/XmlSchema/XmlSchemaSet/TC_SchemaSet_Compile.cs:908]: 👍 @mrj001 thank you for finding these issues and reporting them, as we have an issues tracking them we don't need to keep these comments, even after those issues resolved these comments might forgotten, so please remove them

File: src/libraries/System.Private.Xml/tests/XmlSchema/XmlSchemaSet/TC_SchemaSet_Compile.cs
Comment (buyaa-n) [inline src/libraries/System.Private.Xml/tests/XmlSchema/XmlSchemaSet/TC_SchemaSet_Compile.cs:926]: NIT: empty line, and please remove invalid formatters assert, they wouldn't be put back

File: src/libraries/System.Private.Xml/tests/XmlSchema/XmlSchemaSet/TC_SchemaSet_Compile.cs
Comment (buyaa-n) [inline src/libraries/System.Private.Xml/tests/XmlSchema/XmlSchemaSet/TC_SchemaSet_Compile.cs:952]: Same here and below, please remove

File: src/libraries/System.Private.Xml/src/Resources/Strings.resx
Comment (mrj001) [reply src/libraries/System.Private.Xml/src/Resources/Strings.resx:1723]: @buyaa-n I've been trying to work on all this feedback, but I'm blocked by #34612 / Nuget/Home#7052

I'll get back to these as soon as I can get the test projects to build in Visual Studio again.

Comment (mrj001) [reply src/libraries/System.Private.Xml/src/Resources/Strings.resx:1723]: @buyaa-n I think I've got a work-around.
I've reworded these two:
Sch_MinLengthGtBaseMinLength
Sch_MaxLengthGtBaseMaxLength
- will push when I have finished all requested changes.

File: src/libraries/System.Private.Xml/tests/XmlSchema/XmlSchemaSet/TC_SchemaSet_Compile.cs
Comment (mrj001) [reply src/libraries/System.Private.Xml/tests/XmlSchema/XmlSchemaSet/TC_SchemaSet_Compile.cs:159]: Fixed.  I got thrown off by the name of the corresponding string resource containing "Gt".  I also noticed that I had put "greater than" in the reworded string, and fixed that too.

File: src/libraries/System.Private.Xml/tests/XmlSchema/XmlSchemaSet/TC_SchemaSet_Compile.cs
Comment (mrj001) [reply src/libraries/System.Private.Xml/tests/XmlSchema/XmlSchemaSet/TC_SchemaSet_Compile.cs:184]: Fixed.

File: src/libraries/System.Private.Xml/tests/XmlSchema/XmlSchemaSet/TC_SchemaSet_Compile.cs
Comment (mrj001) [reply src/libraries/System.Private.Xml/tests/XmlSchema/XmlSchemaSet/TC_SchemaSet_Compile.cs:381]: Done.   The casing of both minLength and maxLength has been updated to match their XML facets in both this test and the string resource.

File: src/libraries/System.Private.Xml/tests/XmlSchema/XmlSchemaSet/TC_SchemaSet_Compile.cs
Comment (mrj001) [reply src/libraries/System.Private.Xml/tests/XmlSchema/XmlSchemaSet/TC_SchemaSet_Compile.cs:385]: Yes, it was for checking for the invalid curly braces.  It made my tests fail until I fixed them.

I've removed all the Regex's.

File: src/libraries/System.Private.Xml/tests/XmlSchema/XmlSchemaSet/TC_SchemaSet_Compile.cs
Comment (mrj001) [reply src/libraries/System.Private.Xml/tests/XmlSchema/XmlSchemaSet/TC_SchemaSet_Compile.cs:389]: renamed it to MaxMinLengthBaseLength_Success_TestData.
Should the test method (MaxMinLengthBaseLength_Test) be renamed too?

File: src/libraries/System.Private.Xml/tests/XmlSchema/XmlSchemaSet/TC_SchemaSet_Compile.cs
Comment (mrj001) [reply src/libraries/System.Private.Xml/tests/XmlSchema/XmlSchemaSet/TC_SchemaSet_Compile.cs:587]: Fixed.

File: src/libraries/System.Private.Xml/tests/XmlSchema/XmlSchemaSet/TC_SchemaSet_Compile.cs
Comment (mrj001) [reply src/libraries/System.Private.Xml/tests/XmlSchema/XmlSchemaSet/TC_SchemaSet_Compile.cs:908]: Comments have been removed.

File: src/libraries/System.Private.Xml/tests/XmlSchema/XmlSchemaSet/TC_SchemaSet_Compile.cs
Comment (mrj001) [reply src/libraries/System.Private.Xml/tests/XmlSchema/XmlSchemaSet/TC_SchemaSet_Compile.cs:926]: All the Regex's have been removed.  There is no longer an empty line between the last line of code in the method and the closing curly brace.

File: src/libraries/System.Private.Xml/tests/XmlSchema/XmlSchemaSet/TC_SchemaSet_Compile.cs
Comment (mrj001) [reply src/libraries/System.Private.Xml/tests/XmlSchema/XmlSchemaSet/TC_SchemaSet_Compile.cs:952]: done.

File: src/libraries/System.Private.Xml/tests/XmlSchema/XmlSchemaSet/TC_SchemaSet_Compile.cs
Comment (buyaa-n) [inline src/libraries/System.Private.Xml/tests/XmlSchema/XmlSchemaSet/TC_SchemaSet_Compile.cs:156]: NIT: we don't really add documentation for test methods, normally method names illustrate what it does, and your method name expressing what it testing very well, please remove the comment.

File: src/libraries/System.Private.Xml/tests/XmlSchema/XmlSchemaSet/TC_SchemaSet_Compile.cs
Comment (buyaa-n) [inline src/libraries/System.Private.Xml/tests/XmlSchema/XmlSchemaSet/TC_SchemaSet_Compile.cs:576]: NIT: Please remove

**Review by buyaa-n — APPROVED** [review]
@mrj001 Thank you for your contribution and addressing the coments, left a few NITs, except that the PR looks great

File: src/libraries/System.Private.Xml/tests/XmlSchema/XmlSchemaSet/TC_SchemaSet_Compile.cs
Comment (mrj001) [reply src/libraries/System.Private.Xml/tests/XmlSchema/XmlSchemaSet/TC_SchemaSet_Compile.cs:701]: Comment is fixed.

File: src/libraries/System.Private.Xml/tests/XmlSchema/XmlSchemaSet/TC_SchemaSet_Compile.cs
Comment (mrj001) [reply src/libraries/System.Private.Xml/tests/XmlSchema/XmlSchemaSet/TC_SchemaSet_Compile.cs:156]: Removed.

File: src/libraries/System.Private.Xml/tests/XmlSchema/XmlSchemaSet/TC_SchemaSet_Compile.cs
Comment (mrj001) [reply src/libraries/System.Private.Xml/tests/XmlSchema/XmlSchemaSet/TC_SchemaSet_Compile.cs:576]: Removed.

File: src/libraries/System.Private.Xml/tests/XmlSchema/XmlSchemaSet/TC_SchemaSet_Compile.cs
Comment (buyaa-n) [reply src/libraries/System.Private.Xml/tests/XmlSchema/XmlSchemaSet/TC_SchemaSet_Compile.cs:156]: Somehow still can see it 😄, maybe forgotten?

Comment (mrj001) [reply src/libraries/System.Private.Xml/tests/XmlSchema/XmlSchemaSet/TC_SchemaSet_Compile.cs:156]: Sorry.  I scrolled through the file, deleted two of them, and mistakenly thought that was all....

It's gone now.  And Ctrl+F shows there aren't any more I missed.

**Comment by mrj001** [comment]
@buyaa-n 
This PR currently addresses 10 of the 13 string resources listed in Issue #30218.
The remaining ones are:
Sch_WhiteSpaceRestriction1
Sch_WhiteSpaceRestriction2
Sch_AttributeRestrictionInvalidFromWildcard

Every XSD I came up with to try them triggered some other error.  Coverlet indicates no code coverage of the above in the System.Private.Xml project's tests.  Shall I just reword them?

File: src/libraries/System.Private.Xml/tests/XmlSchema/XmlSchemaSet/TC_SchemaSet_Compile.cs
Comment (buyaa-n) [reply src/libraries/System.Private.Xml/tests/XmlSchema/XmlSchemaSet/TC_SchemaSet_Compile.cs]: No problem at all, and thanks!

**Comment by buyaa-n** [comment]
> Every XSD I came up with to try them triggered some other error.  Coverlet indicates no code coverage of the above in the System.Private.Xml project's tests.  Shall I just reword them?

If there is no code coverage its good to have one, but some area might be even not reachable, could you post the sample xsd/test case you have tried we might find the issue together, otherwise lets just reword them, maybe in another PR? I will merge this PR after all build passes. 

> This PR currently addresses 10 of the 13 string resources listed in Issue #30218.

Thank you so much!

**Comment by buyaa-n** [comment]
> Looks good but I would either change the test to be locale-independent, or (if there's already functional coverage) consider not bothering with a test at all. This would be acceptable for any cases where you are literally just changing the text of the string and there are no replacement parameters in it.

@danmosemsft I think @mrj001 addressed your change request, tests added using schema attribute names instead of entire error messages as string literal, tests looks good and improving code coverage, do you have any other comments/suggestions? 

Tagging also @krwq as he back

**Comment by danmosemsft** [comment]
@buyaa-n don't wait on me, I trust your review

**Comment by buyaa-n** [comment]
> @buyaa-n don't wait on me, I trust your review

Thanks, as you have requested change we still need your approval 😄 for merge

**Review by danmoseley — APPROVED** [review]

**Comment by danmosemsft** [comment]
Oops, done

**Comment by danmosemsft** [comment]
Thank you @mjr001 for the contribution. Perhaps you'd be interested in looking at another issue that is labeled up for grabs.

**Comment by krwq** [comment]
LGTM post merge

**Pull request merged**
